### PR TITLE
Header top nav redesign (support update)

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Adds the new Header Top Nav based on a Header redesign. In order to have a smooth transition between DCR and Frontend, this is kept behind a feature switch named `header-top-nav`.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [X] Yes – https://github.com/guardian/dotcom-rendering/pull/6420

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |
| ![beforeM][] | ![afterM][] |

[before]: https://user-images.githubusercontent.com/76776/201294305-0c6761c3-d6e8-48d2-96e2-2a47586e7f22.png
[after]: https://user-images.githubusercontent.com/76776/201316032-6dde9b9a-9e69-4d09-a28c-ed9458661513.png

[beforeM]: https://user-images.githubusercontent.com/76776/201327555-77f2a9ba-657c-4b8c-84c6-85ae07ee8ac6.png
[afterM]: https://user-images.githubusercontent.com/76776/201327512-b000c15c-404a-4ad8-ad10-8c0c3916cb8e.png

## What is the value of this and can you measure success?

It is urgently needed since we release the new single front door call to action!

### Does this affect other platforms?

- [X] AMP – Sort of, mainly updating the CTA (call to actions) in https://github.com/guardian/dotcom-rendering/pull/6407
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [X] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [X] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [X] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [X] On CODE (optional)

### CODE REVIEW

File: common/app/navigation/DCRNavigation.scala [hunk 1]
```diff
@@ -46,7 +53,7 @@ object ReaderRevenueLinks {
     getReaderRevenueUrl(SupportContribute, SideMenu),
     getReaderRevenueUrl(SupportSubscribe, SideMenu),
     getReaderRevenueUrl(Support, SideMenu),
-    getReaderRevenueUrl(SupporterCTA, SideMenu),
```
Comment (OllysCoding) [inline common/app/navigation/DCRNavigation.scala:49]: Does this have a visual impact on DCR?

Comment (mxdvl) [reply common/app/navigation/DCRNavigation.scala:49]: I think only AMP

**Review by rBangay — APPROVED** [review]

File: common/app/navigation/helpers/UrlHelpers.scala [hunk 2]
```diff
@@ -64,8 +70,13 @@ object UrlHelpers {
...
     List(
-      NavLink("Make a contribution", getReaderRevenueUrl(SupportContribute, SideMenu)),
-      NavLink("Subscribe", getReaderRevenueUrl(SupportSubscribe, SideMenu), classList = Seq("js-subscribe")),
+      NavLink("Support us", getReaderRevenueUrl(SupportContribute, SideMenu)),
+      NavLink(
+        "Print subscriptions",
+        if (Edition(request).id == "UK") getReaderRevenueUrl(SupporterCTA, SideMenu)
+        else getReaderRevenueUrl(PrintCTAWeekly, SideMenu),
+        classList = Seq("js-print-subscription"),
+      ),
```
Comment (OllysCoding) [inline common/app/navigation/helpers/UrlHelpers.scala:79]: Does this get used on the old header & the new header?

File: static/src/stylesheets/layout/nav/_veggie-burger.scss [hunk 3]
```diff
@@ -12,8 +12,9 @@
     z-index: $zindex-main-menu - 1;
 
     // menu is open
-    .new-header--open & {
-        z-index: $zindex-main-menu + 1;
+    .new-header--open &,
```
Comment (OllysCoding) [inline static/src/stylesheets/layout/nav/_veggie-burger.scss:15]: So glad the new-header isn't the new header

File: common/app/views/fragments/headerTopNav.scala.html [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                                views.html.fragments.inlineSvg(icon, "icon", isPresentation = true)
+                                })
+                                @item.label
+                            </a>
+                        </li>
+                        }
+                    </ul>
+                </div>
+
+                @if(editionId != "au" || editionId == "uk") {
```
Comment (AshCorr) [inline common/app/views/fragments/headerTopNav.scala.html:92]:
```suggestion
                @if(editionId != "au") {
```
Surely if `editionId` was `uk` it would already be caught by the `editionID != "au"`?

**Review by OllysCoding — APPROVED** [review]
+1 Left a few questions - I didn't feel like I could meaningfully review the new massive chunks of html, js & css, but to my eyes they look okay!

File: static/src/javascripts/projects/common/modules/navigation/header-top-nav.js [hunk 5]
```diff
@@ -0,0 +1,555 @@
+import debounce from 'lodash/debounce';
```
Comment (AshCorr) [inline static/src/javascripts/projects/common/modules/navigation/header-top-nav.js:1]: Guessing most of this is copied from `new-header.js` which is actually the old header? :)

**Review by AshCorr — APPROVED** [review]
TLDR: Don't name things "new" or "next gen" because eventually they won't be new!

Ditto what Olly said, I reviewed the twirl and sass and didn't see anything immediately glaring!

I'm quite excited for this actually, I really like the look of the new top bar, nicely done Max!

**Review by jamesgorrie — APPROVED** [review]
I like the approach of having a separate header. There's few places we might have split things more explicitly, but it seems like good pragmatic choices. Nice.

I've tried to go through most things not copy/pasted, and all looks great, ran it locally too for good measure. Superb!

File: common/app/navigation/helpers/UrlHelpers.scala (hunk 2 above)
Comment (mxdvl) [reply common/app/navigation/helpers/UrlHelpers.scala:79]: Yes, but on Mobile that’s always what we want.

File: static/src/stylesheets/layout/nav/_veggie-burger.scss (hunk 3 above)
Comment (mxdvl) [reply static/src/stylesheets/layout/nav/_veggie-burger.scss:15]: I almost went with `newest-header` but then thought of future travellers ;)

File: static/src/stylesheets/layout/nav/_pillars.scss [hunk 6]
```diff
@@ -3,7 +3,8 @@
     margin: 0;
     padding: 0 0 0 ($gs-gutter / 2);
 
-    .new-header:not(.new-header--slim) & {
+    .new-header:not(.new-header--slim) &,
+    .header-top-nav:not(.header-top-nav--slim) & {
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/stylesheets/layout/nav/_pillars.scss:7]: Not tested, but it does not happen often on fronts I don’t think.

File: common/app/views/fragments/headerTopNav.scala.html (hunk 4 above)
Comment (mxdvl) [reply common/app/views/fragments/headerTopNav.scala.html:92]: This is how it was… but I see your point.

File: common/app/views/fragments/headerTopNav.scala.html [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+                    <label
+                        class="top-bar__item popup__toggle js-navigation-account-actions js-user-account-trigger is-hidden"
+                        for="my-account-toggle" data-link-name="nav3 : topbar: my account" tabindex="0">
+                        @fragments.inlineSvg("profile", "icon", List("top-bar__item__icon"))
+                        My account
+                    </label>
+                    <div class="my-account__overlay"></div>
+
+                    <ul class="dropdown-menu dropdown-menu--light js-user-account-dropdown-menu"
+                        id="my-account-dropdown" aria-hidden="true">
```
Comment (AshCorr) [inline common/app/views/fragments/headerTopNav.scala.html:73]: Curious as to why this would have aria-hidden?

File: static/src/javascripts/projects/common/modules/navigation/header-top-nav.js (hunk 5 above)
Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/navigation/header-top-nav.js:1]: Yes, exactly! Not confusing at all.

File: common/app/views/fragments/headerTopNav.scala.html [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+                    </a>
+
+                    <input type="checkbox" id="my-account-toggle" aria-controls="my-account-dropdown"
+                        class="dropdown-menu-fallback js-enhance-checkbox" data-link-name="nav3 : topbar: my account"
+                        tabindex="-1" />
+
+                    <label
+                        class="top-bar__item popup__toggle js-navigation-account-actions js-user-account-trigger is-hidden"
+                        for="my-account-toggle" data-link-name="nav3 : topbar: my account" tabindex="0">
+                        @fragments.inlineSvg("profile", "icon", List("top-bar__item__icon"))
```
Comment (AshCorr) [inline common/app/views/fragments/headerTopNav.scala.html:67]: Should this inline svg have `isPresentation = true` so that it is given aria-hidden?

File: common/app/views/fragments/headerTopNav.scala.html [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+            <div class="header-top-nav__top-bar">
+                @defining(Edition(request).id.toLowerCase()) { editionId =>
+                <div class="header-top-nav__item js-print-cta">
+                    <span class="header-top-nav__item--separator hide-until-desktop"></span>
+                    @defining(if (editionId == "uk") getReaderRevenueUrl(PrintCTA,
+                    Header) else getReaderRevenueUrl(PrintCTAWeekly, Header)) { href =>
+                    <a class="top-bar__item hide-until-desktop yellow" data-link-name="nav3 : print-cta"
+                        data-edition="@{editionId}" href="@href">
+                        @fragments.inlineSvg("newspaper", "icon", List("top-bar__item__icon"))
```
Comment (AshCorr) [inline common/app/views/fragments/headerTopNav.scala.html:44]: Should this inline svg have isPresentation = true so that it is given aria-hidden?

File: common/app/views/fragments/headerTopNav.scala.html [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+                    }
+                </div>
+
+
+
+                <div class="header-top-nav__item">
+                    <span class="header-top-nav__item--separator hide-until-desktop"></span>
+                    <a class="top-bar__item js-navigation-sign-in my-account" data-link-name="nav3 : topbar : signin"
+                        href="@Configuration.id.url/signin?INTCMP=DOTCOM_NEWHEADER_SIGNIN&@createAuthenticationComponentEventParams(SigninHeaderId)">
+                        @fragments.inlineSvg("profile", "icon", List("top-bar__item__icon"))
```
Comment (AshCorr) [inline common/app/views/fragments/headerTopNav.scala.html:56]: Should this inline svg have isPresentation = true so that it is given aria-hidden?

File: common/app/views/fragments/headerTopNav.scala.html (hunk 7 above)
Comment (mxdvl) [reply common/app/views/fragments/headerTopNav.scala.html:73]: Because it’s opened with JS. We’ll need to do an accessibility audit as a follow-up because we have not had time to review this properly for the deadline, sadly 😢.

Comment (AshCorr) [reply common/app/views/fragments/headerTopNav.scala.html:73]: Ah it unhides it after its toggled open 👍

File: common/app/navigation/helpers/UrlHelpers.scala (hunk 2 above)
Comment (mxdvl) [reply common/app/navigation/helpers/UrlHelpers.scala:79]: It should also trickle through to AMP. cc @michaelbjacobson

File: common/app/views/fragments/headerTopNav.scala.html (hunk 10 above)
Comment (AshCorr) [reply common/app/views/fragments/headerTopNav.scala.html:56]: Don't think you do need aria-hidden since they're already accompanied by text in the link!

File: static/src/stylesheets/layout/nav/_pillars.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/stylesheets/layout/nav/_pillars.scss:7]: Should have removed these in #25665
